### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-php/pulls) for the same formula update/change?
- [ ] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

-----
Related to https://github.com/Homebrew/homebrew-php/issues/4540

Personally I'd like to go even further than I did here and just drop all mentions of unsupported/outdated PHP and macOS versions. But that probably needs some more discussion.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -58,10 +60,10 @@ end
...
+Defining extensions inheriting AbstractPhp(54,55,56,70,71). Extension will provide a `write_config_file` which add `ext-{extension}.ini` to `conf.d`, don’t forget to remove it manually upon extension removal. Please see [abstract-php-extension.rb](Abstract/abstract-php-extension.rb) for more details.
 
 Please note that your formula installation may deviate significantly from the above; caveats should more or less stay the same, as they give explicit instructions to users as to how to ensure the extension is properly installed.
 
 The ordering of formula attributes, such as the `desc`, `homepage`, `url`, `sha256`, etc. should follow the above order for consistency. The `version` is only included when the URL does not include a version in the filename. `head` installations are not required.
 
-All official PHP extensions should be built for all stable versions of PHP included in `homebrew-php`. These versions are `5.3.29`, `5.4.45`, `5.5.38`, `5.6.27`, and `7.0.12`.
+All official PHP extensions should be built for all stable versions of PHP included in `homebrew-php`. These versions are `5.4`, `5.5`, `5.6`, `7.0`, and `7.1`.
```
Comment (alcaeus) [inline CONTRIBUTING.md]: According to https://secure.php.net/supported-versions.php, only 5.6, 7.0 and 7.1 are supported. Discussion in #3686 revealed that dropping 5.3 and 5.4 is feasible, but 5.5 (and newer, obviously) should be kept around. Not saying how to change it, just summarizing past discussions regarding deprecations.

File: README.md [hunk 2]
```diff
@@ -103,7 +103,7 @@ The purpose of this repository is to allow PHP developers to quickly retrieve wo
 $ brew options php56
```
Comment (alcaeus) [inline README.md]: Just a suggestion: use this opportunity to change commands to the newest version. I've found people are very likely to copy/paste stuff from docs, so we could ease pains by just putting `php71` there for now.

File: CONTRIBUTING.md (hunk 1 above)
Comment (SMillerDev) [reply CONTRIBUTING.md]: Since its still building I kept 5.4 in but I'd be happy to drop it as soon as it breaks.

File: README.md (hunk 2 above)
Comment (SMillerDev) [reply README.md]: Done, I kept the upgrade path from php5 versions though. To also show the difference between loading 5 and 7

**Pull request closed without merging**